model/src/main/kotlin/utils/FileArchiverStorage.kt

```diff
@@ -43,3 +43,12 @@ interface FileArchiverStorage {
      */
     fun getArchive(provenance: Provenance): File?
 }
+
+/**
+ * Checks whether the given [provenance] is empty and throws a runtime exception if it is.
+ */
+internal fun checkNotEmpty(provenance: Provenance) {
```

Comment:
Would it make sense to have this logic in the `Provenance` class itself?

Reply:
Imo not, because it is file archiver specific.